kmod/igb/e1000_82575.c

```diff
@@ -1797,7 +1799,13 @@ static s32 e1000_get_media_type_82575(struct e1000_hw *hw)
 		/* fall through for I2C based SGMII */
 	case E1000_CTRL_EXT_LINK_MODE_PCIE_SERDES:
 		/* read media type from SFP EEPROM */
+#ifdef I2C_ENABLED
+		printk(KERN_INFO "igb_avb I2C enabled - set_sfp_media_type_82575() called");
 		ret_val = e1000_set_sfp_media_type_82575(hw);
+#else
+		printk(KERN_INFO "igb_avb I2C disabled - set_sfp_media_type_82575() not necessary");
+		hw->phy.media_type = e1000_media_type_unknown;
```

Comment:
I agree, no problem. I viewed the larger scope of the function and see this SUCCESS being set.